src/amo/components/CollectionManager/index.js

```diff
@@ -211,12 +213,24 @@ export class CollectionManagerBase extends React.Component<
       creating,
       currentUsername,
       errorHandler,
+      filters,
       i18n,
       isCollectionBeingModified,
+      showEditForm,
       siteLang,
     } = this.props;
     const { description, name, slug } = this.state;
 
+    if (!showEditForm) {
+      return (
+        <CollectionDetails
+          collection={collection}
+          filters={filters}
+          showEditButton={false}
```

Comment:
Once you enter edit mode as a maintainer, there is no way to get out of edit mode (besides pressing the browser *back* button). I suggest showing a button like *Finish editing* which will behave the same as *Cancel* on the edit form.